lib/routemap.c

```diff
@@ -1281,14 +1292,39 @@ const char *route_map_get_match_arg(struct route_map_index *index,
 	return (NULL);
 }
 
+static int8_t get_route_map_delete_event(route_map_event_t type)
+{
+	switch (type) {
+	case RMAP_EVENT_CALL_ADDED:
+		return RMAP_EVENT_CALL_DELETED;
+	case RMAP_EVENT_PLIST_ADDED:
+		return RMAP_EVENT_PLIST_DELETED;
+	case RMAP_EVENT_CLIST_ADDED:
+		return RMAP_EVENT_CLIST_DELETED;
+	case RMAP_EVENT_ECLIST_ADDED:
+		return RMAP_EVENT_ECLIST_DELETED;
+	case RMAP_EVENT_LLIST_ADDED:
+		return RMAP_EVENT_LLIST_DELETED;
+	case RMAP_EVENT_ASLIST_ADDED:
+		return RMAP_EVENT_ASLIST_DELETED;
+	case RMAP_EVENT_FILTER_ADDED:
+		return RMAP_EVENT_FILTER_DELETED;
+	default:
```

Comment:
Please remove default from a switch statement that looks at an enum

Reply:
Thanks, @donaldsharp.
Removing the 'default' leads to compilation warnings.
Instead, will it be okay to introduce assert() within 'default'?
I can rectify the caller of this function to pass the right enum values as an argument to this function.

Reply:
perhaps you can add all the other enum types with an explanation that nothing should be expected here for these values and why?

Reply:
Sure, I can do that. Thanks @donaldsharp.

Reply:
I've included the other enum values and a default case where the code asserts. I've provided a note on why the code asserts.